app/src/main/java/org/fossasia/openevent/general/ticket/TicketsViewModel.kt

```diff
@@ -122,23 +130,44 @@ class TicketsViewModel(
                 val code = appliedDiscountCode
                 tickets.forEach { ticket ->
                     var price = ticket.price
+                    totalTaxAmount += (ticket.price * taxRate / 100) * qty[index]
                     if (code?.value != null) {
                         appliedDiscountCode?.tickets?.forEach { ticketId ->
                             if (ticket.id == ticketId.id.toInt()) {
                                 price -= if (code.type == AMOUNT) code.value else price*(code.value / 100)
                             }
                         }
                     }
-                    price.let { prices += price * qty[index++] }
+                    price.let { prices += price * qty[index] }
```

Comment:
Unnecessary change?

Reply:
Not unnecessary, the index is updated at the end.